README.md

```diff
@@ -227,3 +234,26 @@ following commands from `/lib` to build docs.
 * `./gradlew dokkaHtmlJava` to build java format docs
 
 Generated documentation can be found in their respective folders inside `/build/dokka`.
+
+## Release checklist
+Perform these actions before releasing a new starknet-jvm version:
+1. Checkout to `main` and pull
+```
+git checkout main && git pull
+```
+2. Create new branch for version bump
+```
+git checkout -b chore/bump-version-to-0.x.x
+```
+3. Update the version in `lib/build.gradle.kts` (following [semantic versioning](https://semver.org/)).
+Note: We will never increment MAJOR version.
```

Comment:
I think we can get rid of this note :) alternatively you can say that MINOR or PATCH should be incremented for most changes or sth